airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -250,8 +259,19 @@ private static Runnable readFromDstRunnable(final AirbyteDestination destination
             throw new DestinationException("Destination process read attempt failed", e);
           }
           if (messageOptional.isPresent()) {
-            LOGGER.info("State in DefaultReplicationWorker from destination: {}", messageOptional.get());
-            messageTracker.acceptFromDestination(messageOptional.get());
+            final AirbyteMessage message = messageOptional.get();
+            LOGGER.info("State in DefaultReplicationWorker from destination: {}", message);
+            messageTracker.acceptFromDestination(message);
+
+            try {
+              if (message.getType() == Type.CONTROL && message.getControl().getType() == AirbyteControlMessage.Type.CONNECTOR_CONFIG) {
+                persistConfigHelper.persistDestinationConfig(jobId, message.getControl().getConnectorConfig().getConfig());
+              }
+            } catch (final ApiException e) {
+              LOGGER.error("Error trying to save updated destination config", e);
+              throw new RuntimeException(e);
+            }
```

Comment:
👍 that if we can't persist a config update we should fail the sync.
As a note, I read this as a blocking activity - processing updated configs will block all other message processing.  This is probably fine as it won't happen that often in the sync